app/bundles/PluginBundle/Integration/AbstractIntegration.php

```diff
@@ -1133,6 +1133,110 @@ public function populateLeadData($lead, $config = array())
     }
 
     /**
+     * Takes profile data from an integration and maps it to Mautic's lead fields
+     *
+     * @param       $data
+     * @param array $config
+     *
+     * @return array
+     */
+    public function populateMauticLeadData($data, $config = array())
+    {
+        // Glean supported fields from what was returned by the integration
+        $gleanedData = $this->matchUpData($data);
+
+        if (!isset($config['leadFields'])) {
+            $config = $this->mergeConfigToFeatureSettings($config);
+
+            if (empty($config['leadFields'])) {
+
+                return array();
+            }
+        }
+
+        $leadFields      = $config['leadFields'];
+        $availableFields = $this->getAvailableLeadFields($config);
+        $matched         = array();
+        $this->factory->getLogger()->addError(print_r($leadFields, true)); 
+        $this->factory->getLogger()->addError(print_r($gleanedData, true));
+
+
+        foreach ($gleanedData as $key => $field) {
+            if (isset($leadFields[$key]) && isset($gleanedData[$key])) {
+                $matched[$leadFields[$key]] = $gleanedData[$key];
+            }
+        }
+        $this->factory->getLogger()->addError(print_r($matched, true));
+        return $matched;
+    }
+
+    /**
+     * Create or update existing Mautic lead from the integration's profile data
+     *
+     * @param mixed     $data     Profile data from integration
+     * @param bool|true $persist  Set to false to not persist lead to the database in this method
+     *
+     * @return Lead
+     */
+    public function getMauticLead($data, $persist = true)
+    {
+        if (is_object($data)) {
+            // Convert to array in all levels
+            $data = json_encode(json_decode($data));
```

Comment:
This would return an object, wouldn't it? Maybe it should be
```
$data = json_encode(json_decode($data), true);
```
Although I didn't test it...